### PR TITLE
Backport of Stuck-Saving

### PR DESCRIPTION
**Pull request by Ashod**
- wsd: test: handle save notifications in tests
- wsd: test: filterSendMessage -> filterSendWebSocketMessage
- wsd: test: cleanup and cosmetics
- wsd: Message supports new find and contains operations
- wsd: test: support filtering Kit messages
- wsd: timeout on saving stuck with test

### CODE REVIEW

**Review by thebearon — APPROVED** [review]
Thanks Ash!

File: wsd/DocumentBroker.cpp [hunk 1]
```diff
@@ -1787,6 +1795,9 @@ void DocumentBroker::setLoaded()
         _docState.setLive();
         _loadDuration = std::chrono::duration_cast<std::chrono::milliseconds>(
                                 std::chrono::steady_clock::now() - _threadStart);
+        _saveManager.setSavingTimeout(
```
Comment (mmeeks) [inline wsd/DocumentBroker.cpp:1798]: Not convinced by this here.

**Review by mmeeks — COMMENTED** [review]
Not sure that the arithmetic here is good around saving and the time it should take.
I can imagine a document loading very fast say 50ms and taking 10 seconds to save on a busy system. No data on that - but not sure we cope with that well.

What's the story there - why 5 seconds ? and so on =)

File: wsd/DocumentBroker.cpp (hunk 1 above)
Comment (Ashod) [reply wsd/DocumentBroker.cpp:1798]: It's a way to cap it, with the intent to improve it and tune it. Agreed that it's a bit finger-in-the-air formula, but being able to cap the wait prevents getting stuck. Of course adding tracking of save duration and using that would be better. (We could also make it configurable, but we have too many of those already.)

**Pull request merged**